code/src/components/CustomCheckbox.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+
+// normal functional comp
+export const CustomCheckbox = () => {
+  const [checked, setChecked] = useState(false);
+
+  //  this is a functional component in a component, custom made checkbox
+  const Checkbox = ({ className, checked, ...props }) => (
+    <CheckboxContainer>
+      <HiddenCheckbox checked={checked} {...props}></HiddenCheckbox>
+      <StyledCheckbox checked={checked}>
+        <Svg viewBox="0 0 24 24">
+          <polyline points="20 3 9 17 4 11" />
+        </Svg>
+      </StyledCheckbox>
+    </CheckboxContainer>
+  );
+
+  const handleOnChange = (event) => {
+    setChecked(event.target.checked);
+  };
+
+  const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
+    border: 0;
+    clip: rect(0 0 0 0);
+    clippath: insert(50%);
+    height: 1px;
+    margin: -1px;
+    overflow: hidden;
+    padding: 0;
+    position: absolute;
+    white-space: nowrap;
+    width: 1px;
+  `;
+
+  //  The check mark, is defined here
+  const Svg = styled.svg`
```

Comment:
The cool thing with styled components is that we can pick more descriptive names for the HTML tags ;)